sdk/tests/conformance/extensions/oes-fbo-render-mipmap.html

```diff
@@ -0,0 +1,92 @@
+<!DOCTYPE html>
+<html>
+<head>
+<meta charset="utf-8">
+<title>WebGL OES_fbo_render_mipmap Conformance Tests</title>
+<link rel="stylesheet" href="../../resources/js-test-style.css"/>
+<script src="../../js/js-test-pre.js"></script>
+<script src="../../js/webgl-test-utils.js"></script>
+</head>
+<body>
+<canvas id="canvas" width="128" height="128"></canvas>
+<div id="description"></div>
+<div id="console"></div>
+<script>
+"use strict";
+description("This test verifies the functionality of the OES_fbo_render_mipmap extension, if it is available.");
+
+var wtu = WebGLTestUtils;
+var canvas = document.getElementById("canvas");
+var gl = wtu.create3DContext(canvas);
+var ext = null;
+
+(function(){
+
+    const oesFboRenderMipmap = gl.getExtension("OES_fbo_render_mipmap");
+    if (!oesFboRenderMipmap) {
+        testPassed("OES_fbo_render_mipmap is allowed to be missing.");
+        return;
+    }
+
+    let texture = gl.createTexture();
+    gl.bindTexture(gl.TEXTURE_2D, texture);
+    gl.texParameteri(gl.TEXTURE_2D, gl.TEXTURE_MAG_FILTER, gl.NEAREST);
+    gl.texParameteri(gl.TEXTURE_2D, gl.TEXTURE_MIN_FILTER, gl.NEAREST_MIPMAP_NEAREST);
+
+    let colors = [
+        [255, 0, 0, 255],
+        [255, 127, 0, 255],
+        [255, 255, 0, 255],
+        [0, 255, 0, 255],
+        [0, 255, 255, 255],
+        [0, 127, 255, 255],
+        [0, 0, 255, 255],
+        [255, 0, 255, 255],
+    ];
+
+    let fbos = [];
+    let maxLevel = 7;
+    for (let level = 0; level <= maxLevel; level++) {
+        let size = Math.pow(2, maxLevel - level);
+        gl.texImage2D(gl.TEXTURE_2D, level, gl.RGBA, size, size, 0, gl.RGBA, gl.UNSIGNED_BYTE, null);
+        let fbo = gl.createFramebuffer();
+        gl.bindFramebuffer(gl.FRAMEBUFFER, fbo);
+        gl.framebufferTexture2D(gl.FRAMEBUFFER, gl.COLOR_ATTACHMENT0, gl.TEXTURE_2D, texture, level);
+        fbos.push(fbo);
+
+        wtu.glErrorShouldBe(gl, gl.NO_ERROR, "Any level of a texture can be attached to a framebuffer object.");
+    }
+
+    shouldBe("gl.checkFramebufferStatus(gl.FRAMEBUFFER)", "gl.FRAMEBUFFER_COMPLETE");

```

Comment:
Do this per FB, or not at all.

Reply:
Thanks for the review. I write this test against chromium with roughly hooked up angle impl of this functionality. Turns out checking framebuffer completeness before all mip levels of the texture is created will ends in FRAMEBUFFER_INCOMPLETE_ATTACHMENT.

I just checked the OpenGL ES 2 Spec https://www.khronos.org/registry/OpenGL/specs/es/2.0/es_cm_spec_2.0.pdf
And seems like it makes sense
> Performing any of the following actions may change whether the framebuffer is considered complete or
incomplete.
> • Changing the width, height, or internal format of a texture image that is attached to the framebuffer
by calling TexImage2D, CopyTexImage2D and CompressedTexImage2D.
> This means that rendering commands such as DrawArrays, DrawElements, any command that reads the framebuffer such as ReadPixels and CopyTexSubImage will generate the
error INVALID FRAMEBUFFER OPERATION if called while the framebuffer is not framebuffer complete

But it should be checked for every fbo still. So I put into the second loop (together with the clear command)

Reply:
Just test with Firefox Nightly 72 on Linux. Proved the assumption. It wouldn't work if checking framebuffer completeness and calling gl.clear before all mip levels are created.